diff -urN a/nvidia-persistenced/nvidia-persistenced.c b/nvidia-persistenced/nvidia-persistenced.c
--- a/nvidia-persistenced/nvidia-persistenced.c	2020-06-25 06:00:27 +0900
+++ b/nvidia-persistenced/nvidia-persistenced.c	2021-01-08 07:06:41 +0900
@@ -689,7 +689,7 @@
  */
 static int daemonize(uid_t uid, gid_t gid)
 {
-    char pid_str[10];
+    char pid_str[13];
     struct sigaction signal_action;
     sigset_t signal_set;
     int init_pipe_fds[2];
