diff -urN a/kernel/common/inc/nv-linux.h b/kernel/common/inc/nv-linux.h
--- a/kernel/common/inc/nv-linux.h	2021-07-28 13:50:30 +0900
+++ b/kernel/common/inc/nv-linux.h	2021-07-28 13:59:14 +0900
@@ -168,6 +168,9 @@
 
 #include <linux/pagemap.h>
 #include <linux/dma-mapping.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+#include <linux/dma-map-ops.h>
+#endif
 
 #if defined(CONFIG_SWIOTLB) && defined(NVCPU_AARCH64)
 #include <linux/swiotlb.h>
diff -urN a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2021-07-28 13:50:30 +0900
+++ b/kernel/conftest.sh	2021-07-28 13:58:32 +0900
@@ -1473,6 +1473,10 @@
             #
             CODE="
             #include <linux/dma-mapping.h>
+            #include <linux/version.h>
+            #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+            #include <linux/dma-map-ops.h>
+            #endif
             void conftest_get_dma_ops(void) {
                 get_dma_ops();
             }"
diff -urN a/kernel/nvidia-drm/nvidia-drm-gem-user-memory.c b/kernel/nvidia-drm/nvidia-drm-gem-user-memory.c
--- a/kernel/nvidia-drm/nvidia-drm-gem-user-memory.c	2020-05-29 17:38:06 +0900
+++ b/kernel/nvidia-drm/nvidia-drm-gem-user-memory.c	2021-07-28 14:08:34 +0900
@@ -47,8 +47,14 @@
 {
     struct nv_drm_gem_user_memory *nv_user_memory = to_nv_user_memory(nv_gem);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0)
+    return drm_prime_pages_to_sg(nv_user_memory->base.nv_dev->dev,
+                                 nv_user_memory->pages,
+                                 nv_user_memory->pages_count);
+#else
     return drm_prime_pages_to_sg(nv_user_memory->pages,
                                  nv_user_memory->pages_count);
+#endif
 }
 
 static void *__nv_drm_gem_user_memory_prime_vmap(
