diff -urN a/kernel/common/inc/nv-linux.h b/kernel/common/inc/nv-linux.h
--- a/kernel/common/inc/nv-linux.h	2020-05-29 17:38:07 +0900
+++ b/kernel/common/inc/nv-linux.h	2021-07-28 12:46:55 +0900
@@ -509,7 +509,12 @@
 
 static inline void *nv_vmalloc(unsigned long size)
 {
-    void *ptr = __vmalloc(size, GFP_KERNEL, PAGE_KERNEL);
+    void *ptr;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
+    ptr = __vmalloc(size, GFP_KERNEL);
+#else
+    ptr = __vmalloc(size, GFP_KERNEL, PAGE_KERNEL);
+#endif
     if (ptr)
         NV_MEMDBG_ADD(ptr, size);
     return ptr;
@@ -1818,6 +1823,10 @@
 #include <linux/backlight.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
+#define mmap_sem mmap_lock
+#endif
+
 #define MODULE_BASE_NAME "nvidia"
 #define MODULE_INSTANCE_NUMBER 0
 #define MODULE_INSTANCE_STRING ""
